packages/core/src/utils/public.ts

```diff
@@ -321,3 +322,23 @@ export function shuffle(array: any[]): void {
 export function generateId(): string {
   return nanoid();
 }
+
+/**
+ * parse plugin name, return the plugin protocol and url object
+ */
+export function parsePluginName(name: string): { protocol: PluginProtocol; urlObject: url.UrlWithStringQuery } {
```

Comment:
public_utils is mostly for the plugin developer, so IMO this function should be another utils module.

Reply:
OK, should we create a new utils package?

Reply:
No new package, they could be put in core/src/utils.

Reply:
OK.